File: Module 1/C# II - Advanced/homework_2_c_sharp_due_25.11.2016/05. Larger than neighbours/LargerThanNeighbours.cs
/*
Larger than neighbours

Description
Write a method that checks if the element at given position in given array of integers 
is larger than its two neighbours (when such exist). 
Write program that reads an array of numbers and prints how many of them are larger than their neighbours.

Input
On the first line you will receive the number N - the size of the array
On the second line you will receive N numbers separated by spaces - the array

Output
Print how many numbers in the array are larger than their neighbours

Constraints
1 <= N <= 1024
Time limit: 0.1s
Memory limit: 16MB

Sample tests

Input
6
-26 -25 -28 31 2 27

Output
2
*/

using System;

class LargerThanNeighbours
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        string[] input = Console.ReadLine().Split(' ');
        int[] array = new int[n];
        for (int i = 0; i < n; i++)
        {
            array[i] = int.Parse(input[i]);
        }
        int result = 0;
        for (int i = 0; i < n; i++)
        {
            if (IsLargerThanNeighbours(array, i)) result++;
        }

        Console.WriteLine(result);
    }

    private static bool IsLargerThanNeighbours(int[] array, int position)
    {
        if (position <= 0 || position >= array.Length - 1)
        {
            return false;
        }
        else
        {
            if (array[position] > array[position + 1] && array[position] > array[position - 1])
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: Module 1/JS I - Fundamentals/homework_5_due_13.01.2017/problem_05/LargerThanNeighbours.js
// Problem 05. Larger than Neighbours - BGCoder format
function solve(args) {
	let count = +args[0],
		arr = args[1].split(' ').map(Number);
	
	function LargerThanNeighbours(arr, i) {
		return ((arr[i] > arr[i - 1]) && (arr[i] > arr[i + 1]));
	}
	
	function CountAppearances(arr, len) {
		let result = 0;
		
		for (let i = 1; i < count; i += 1) {
			if (LargerThanNeighbours(arr, i)) {
				result += 1;
			}
		}
		
		return (result);
	}
	
	let result = CountAppearances(arr, count);
	
	return (result);
}